toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -486,23 +463,20 @@ class ErrorTrackingDiagnosticConsumer : public DiagnosticConsumer {
 // given emitter. That function can annotate the diagnostic by calling
 // `builder.Note` to add notes.
 template <typename LocationT, typename AnnotateFn>
-class DiagnosticAnnotationScope
-    : private DiagnosticEmitter<LocationT>::DiagnosticAnnotationScopeBase {
-  using Base =
-      typename DiagnosticEmitter<LocationT>::DiagnosticAnnotationScopeBase;
+class DiagnosticAnnotationScope {
+ private:
+  using Emitter = DiagnosticEmitter<LocationT>;
```

Comment:
You might consider something like making the private `using` public because it's a bit inconsistent with style to have two private sections (https://google.github.io/styleguide/cppguide.html#Class_Format, "The public section should be first" etc)

Reply:
I don't think it makes sense to add undesirable things to the class's public API to work around a style rule. I've just duplicated this type into the two places it was used, but I think this was better before. *grumble grumble*.